bootstrap/cmd/bootstrap/app/gcpUtils.go

```diff
@@ -187,11 +196,18 @@ func (s *ksServer)ApplyIamPolicy(ctx context.Context, req ApplyIamRequest) error
 
 	// Get the updated policy and apply it.
 	newPolicy := GetUpdatedPolicy(saPolicy, &iamConf, req)
-	_, err = resourceManager.Projects.SetIamPolicy(
-		req.Project,
-		&cloudresourcemanager.SetIamPolicyRequest{
-			Policy: &newPolicy,
-		}).Do()
+	retry = 0
+	if retry < 5 {
+		_, err = resourceManager.Projects.SetIamPolicy(
```

Comment:
If the failure is caused by concurrent iam policy change, we need to re-get the changed iam policy as refresh start and then apply configs to it and try to set iam again.

Reply:
I see, updated. Thanks